tensorflow/python/ops/image_ops_impl.py

```diff
@@ -511,6 +512,77 @@ def _rot270():
   result.set_shape([shape[0], None, None, shape[3]])
   return result
 
+@tf_export('image.rot90ND')
+def rot90ND(image, rot_dims, k=1, name=None):
+  """Rotate image(s) counter-clockwise by 90 degrees.
+
+  Args:
+    image: N-D Tensor where N >= 3 
+    k: A scalar integer. The number of times the image is rotated by 90 degrees.
+    rot_dims: A list of integers with length 2. The dimensions to be rotated. 
+    name: A name for this operation (optional).
+
+  Returns:
+    A rotated tensor of the same type and shape as `image`.
+
+  Raises:
+    ValueError: if the shape of `image` not supported.
+  """
+
+  if len(rot_dims) != 2:
+    raise ValueError("rot_dims must be exactly length 2 "
+                     "(length: {})".format(len(rot_dims)))
+  rot_dims.sort()
+
+  shape = image.get_shape()
+  if shape.ndims is None:
+    ndims = 3
```

Comment:
Is it safe to assume that unknown rank tensors are 3 dimensional? Can't that be inconsistent with the values in `rot_dims`? `ndims` must be at least the max of the values in `rot_dims`? Also, what error would one get if this inference is incorrect? For example, feeding a 2D array or a 5D array?